server/functions/nextval.go

```diff
@@ -24,18 +24,23 @@ import (
 
 // initNextVal registers the functions to the catalog.
 func initNextVal() {
-	framework.RegisterFunction(nextval_text)
+	framework.RegisterFunction(nextval_regclass)
 }
 
-// nextval_text represents the PostgreSQL function of the same name, taking the same parameters.
-var nextval_text = framework.Function1{
+// nextval_regclass represents the PostgreSQL function of the same name, taking the same parameters.
+var nextval_regclass = framework.Function1{
 	Name:               "nextval",
 	Return:             pgtypes.Int64,
-	Parameters:         [1]pgtypes.DoltgresType{pgtypes.Text},
+	Parameters:         [1]pgtypes.DoltgresType{pgtypes.Regclass},
 	IsNonDeterministic: true,
 	Strict:             true,
 	Callable: func(ctx *sql.Context, _ [2]pgtypes.DoltgresType, val any) (any, error) {
-		schema, sequence, err := parseRelationName(ctx, val.(string))
+		relationName, err := pgtypes.Regclass.IoOutput(ctx, val)
```

Comment:
Ahh... that would have been a pretty helpful design decision to include in a comment in that code! 🤣 I'll add that comment in my changes. 

Ultimately, we do need to support `nextval()` accepting a `regclass` instance, but we don't need to make it the only path into `nextval()`. It would be very trivial to have an overload for `text` and an overload for `regclass` and at least we'd get faster perf for `text` instances, and not go through the perf overhead of converting to `regclass` for that case. 

Great idea to bring this up in today's sync. If there's a clever way to improve the perf for `regclass` conversion, that could also be a great next step.

Reply:
To close the loop on performance... I ran the Doltgres benchmarks against main, the `nextval(regclass)` implementation, and the overloaded `nextval(text)` / `nextval(regclass)` implementation and you are correct that without the `nextval(text)` overload, the OLTP insert benchmark multiplier in particular goes from 3.2 to 5.3, so indeed, a big perf hit! Once I added in the `nextval(text)` overload, the benchmark multiplier remains at 3.2, and other benchmarks match the results on main. Based on that, the overload is working as expected and keeping the same performance.